e2e/container/run_test.go

```diff
@@ -149,3 +150,55 @@ func TestRunWithCgroupNamespace(t *testing.T) {
 		"/bin/grep", "-q", "':memory:/$'", "/proc/1/cgroup")
 	result.Assert(t, icmd.Success)
 }
+
+func TestMountSubvolume(t *testing.T) {
+	environment.SkipIfAPIOlder(t, "1.45")
```

Comment:
Note: This test will be skipped currently (even with https://github.com/docker/cli/pull/4839) because the 25.0 is still on API `1.44`.

I ran the tests (🟢) locally by substituting the dind image with:
```Dockerfile
FROM docker:25-dind
COPY --from=moby/moby-bin:master /dockerd /usr/local/bin/dockerd
```

<details>

<summary>Tests output</summary>

```
$ make -f docker.Makefile test-e2e
...
✓  e2e/cli-plugins (15.291s)
∅  e2e/cli-plugins/plugins/badmeta
∅  e2e/cli-plugins/plugins/nopersistentprerun
∅  e2e/cli-plugins/plugins/presocket
∅  e2e/internal/fixtures
✓  e2e/global (71ms)
✓  e2e/context (132ms)
∅  e2e/plugin/basic
✓  e2e/plugin (112ms)
✓  e2e/system (37ms)
✓  e2e/stack (264ms)
✓  e2e/trust (1.974s)
✓  e2e/container (5.198s)
✓  e2e/image (10.05s)

=== Skipped
=== SKIP: e2e/cli-plugins TestCLIPluginDialStdio (0.00s)
    dial_test.go:16: skipping plugin dial-stdio test since DOCKER_CLI_PLUGIN_USE_DIAL_STDIO is not set

=== SKIP: e2e/container TestRunAttachedFromRemoteImageAndRemove (0.01s)
    run_test.go:24: daemonPlatform != platform: running against a non linux/amd64 daemon

=== SKIP: e2e/container TestRunWithCgroupNamespace (0.06s)
    run_test.go:147: !cgroupNsFound: running against a daemon that doesn't support cgroup namespaces (security options: [name=seccomp,profile=builtin]
        )

=== SKIP: e2e/image TestPullWithContentTrust (0.01s)
    pull_test.go:22: daemonPlatform != platform: running against a non linux/amd64 daemon

=== SKIP: e2e/image TestPushAllTags (0.01s)
    push_test.go:37: daemonPlatform != platform: running against a non linux/amd64 daemon

=== SKIP: e2e/image TestPushWithContentTrust (0.01s)
    push_test.go:59: daemonPlatform != platform: running against a non linux/amd64 daemon

=== SKIP: e2e/plugin TestInstallWithContentTrust (0.00s)
    trust_test.go:26: skipping until a fix is implemented on moby

=== SKIP: e2e/trust TestSignLocalImage (0.01s)
    sign_test.go:25: daemonPlatform != platform: running against a non linux/amd64 daemon

=== SKIP: e2e/trust TestSignWithLocalFlag (0.01s)
    sign_test.go:43: daemonPlatform != platform: running against a non linux/amd64 daemon
```
</details>